mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1410,6 +1410,39 @@ bool arith::ExtFOp::areCastCompatible(TypeRange inputs, TypeRange outputs) {
 
 LogicalResult arith::ExtFOp::verify() { return verifyExtOp<FloatType>(*this); }
 
+struct SimplifyExtFTruncFOpPair : public OpRewritePattern<ExtFOp> {
```

Comment:
This feels like it could be a folder?

Reply:
use folder instead. BTW, Could you please clarify what's the difference between folder and canonicalizer?

Reply:
Folders run much more often and are meant to be for things that are like constant-folding: fast, local, and unconditionally useful.

Reply:
Folder aren't allowed to create new IR, see doc: https://mlir.llvm.org/docs/Canonicalization/#canonicalizing-with-the-fold-method